src/components/gridList/useGridLayout.ts

```diff
@@ -0,0 +1,49 @@
+import {useMemo} from 'react';
+import {Spacings} from 'style';
+import {useOrientation} from 'hooks';
+import {Constants} from '../../commons/new';
+import {GridListBaseProps} from './types';
+
+export const DEFAULT_NUM_COLUMNS = 3;
+export const DEFAULT_ITEM_SPACINGS = Spacings.s4;
+
+const useGridLayout = (props: GridListBaseProps) => {
+  const {
+    numColumns = DEFAULT_NUM_COLUMNS,
+    itemSpacing = DEFAULT_ITEM_SPACINGS,
+    maxItemWidth,
+    listPadding = 0,
+    keepItemSize,
+    containerWidth
+  } = props;
+
+  const {orientation} = useOrientation();
+
+  const _containerWidth = useMemo(() => {
+    return (containerWidth ?? Constants.screenWidth) - listPadding * 2;
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [listPadding, orientation, containerWidth]);
+
+  const numberOfColumns = useMemo(() => {
+    if (maxItemWidth) {
+      return Math.ceil((_containerWidth + itemSpacing) / (maxItemWidth + itemSpacing));
```

Comment:
Is our interest here is (1) to maximize the number of columns or (2) to maximize the width of each item (and keep it less than the `maxItemWidth`)? 
if option 2 is the answer (and as far as I understand it is), we should use `Math.floor `here and not `Math.ceil`

Reply:
The idea is to max the item width to the maxItemWidth passed by the user. 
When I change to Math.floor I see that items width exceed the maxItemWidth passed in the example screen (I pass 140 and the items width become 169)